CMakeLists.txt

```diff
@@ -155,8 +160,12 @@ pkg_check_modules(ASSIMP REQUIRED assimp)
 pkg_check_modules(SIGCPP REQUIRED sigc++-2.0)
 pkg_check_modules(VORBISFILE REQUIRED vorbisfile)
 
+add_subdirectory(contrib/lz4)
+add_subdirectory(contrib/entt)
```

Comment:
...not used (yet)?

Reply:
Nope, that came in accidentally; I'll fix that up with the next commit.

Reply:
Np. Just a question: are you working on that? Because if you start using that too early, it may make things more difficult to manage as current implementation is incomplete and not clear.

Reply:
I'm using some foundational features of entt (families, signal handlers) internally in prototyping, but I haven't seriously started moving things over yet; I'm working on getting a better grasp of the internals before I do.

Reply:
Got it. If I may share a though, what I would see (or try by myself...) is a change here:
https://github.com/pioneerspacesim/pioneer/blob/be3d9bed7d0826ac50e2fe5d925c7a643f8dbeb5/src/collider/CollisionSpace.h#L57-L58

...and here:

https://github.com/pioneerspacesim/pioneer/blob/be3d9bed7d0826ac50e2fe5d925c7a643f8dbeb5/src/Space.h#L111-L112

I don't think entt "as a whole" can be the answer, but its [sparse set](https://github.com/skypjack/entt/blob/master/src/entt/entity/sparse_set.hpp) or a just little veil over `std::vector`, in order to allow erasing things with "swap&pop", may lead to big improvements without a big investment in coding, isn't?

Reply:
"big improvements" only exist if the code is actually required to be performant. The things you just linked too don't appear to be used in performance critical ways.
Only optimise where optimisation is needed.

Reply:
Ahahah, well, "big improvements" is a `__reserved` concept :wink: ... I just note that a `std::list` of pointers is the least cache friendly structure to use and, actually, it is used by the "main" data structures over the codebase, which are `Body` and `Geoms`.

Thus for bodies any iteration on the whole things means picking objects from here and there in memory (think at `StaticUpdate()`; `Update()`; `Render()` ecc...);

For `Geoms` probably you're right: there's also the `BVHTree`. Please forgive if I'm wrong but I'm new to that code.
However I found at least a "for" which iterate on the whole lists:
https://github.com/pioneerspacesim/pioneer/blob/be3d9bed7d0826ac50e2fe5d925c7a643f8dbeb5/src/collider/CollisionSpace.cpp#L390

> Only optimise where optimisation is needed.

Agreed: to me, after the improvements on PiGui, there's no need for further optimizations, but I feel that these changes, if any further optimisation may become necessary, are "on top" of my list.

Reply:
NOTE: 
>Agreed: to me, after the improvements on PiGui, there's no need for further optimizations, but I feel that these changes, if any further optimisation may become necessary, are "on top" of my list.

@Web-eWorks , the above doesn't mean that this PR is not welcome, obviously... Just that my brain was out of order :smiley: